### PR TITLE
invited Fintrospect blog post

### PR DESCRIPTION
**Pull request by daviddenton**
as suggested by vkostyukov

:)

### CODE REVIEW

**Comment by mosesn** [comment]
Hey, do you mind squashing the commit?

File: source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+---
+
+# Reusable HTTP contracts with Fintrospect
+
+## overview 
+At [SpringerNature][springernature], we've been migrating our architecture stack to a Microservice-based approach, and from the application side a few core requirements kept coming up. We wanted our apps to be able to easily:
+
+- generate and serve live API contract documentation - in our case [Swagger][swagger] docs with [JSON schema][jsonschema] example
+- automatically validate and respond to invalid requests (missing or invalid parameters)
+- marshall objects to and from HTTP requests in various messaging formats in a type-safe way, for both server and client-side services
```
Comment (mosesn) [inline source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md]: s/marshall/marshal

File: source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  .taking(employeeName)
+  .at(Get) / "search" / departmentId
+```
+
+Body parameters are handled in much the same way... with built in support for both JSON and XML.
+
+## serving the contract
+
+#### basics
+In order to serve the contract, we need to bind it to a Finagle ```Service```. Dynamic Path parameters (in this case ```departmentId```) are read from the URL and injected on a request-by-request basis. Other parameters can be retrieved in a type-safe way through applying the ```<--``` operator to the request. Bad requests (missing/malformed parameters) have already been dealt with automatically by this point, so no validation logic is required:
```
Comment (mosesn) [inline source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md]: I think we only need one backtick here, not three?

File: source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+```scala
+import Argonaut.ResponseBuilder._
+import Argonaut.JsonFormat._
+
+case class Employee(name: String)
+
+val resp: ResponseBuilder[Json] = Ok(encode(Employee("Bob"))).withHeaders("MyCustomHeader" -> "value")
+```
+
+The first call to ```Ok``` in the example above is referencing the Finagle ```Status.Ok``` object which is implicitly converted to a ```ResponseBuilder```. We also get implicit conversion from the ```ResponseBuilder``` to a ```Future[Response]```, so an implementation of the server method above could simply be:
```
Comment (mosesn) [inline source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md]: feel free to ignore: s/is referencing/references

File: source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  "com.twitter" %% "finagle-http" % "6.33.0"
+)
+```
+
+
+[springernature]: http://www.springernature.com
+[fintrospect]: http://fintrospect.io
+[supportedjson]: http://fintrospect.io/installation
+[jsonschema]: http://json-schema.org/
+[demo]: http://github.com/daviddenton/fintrospect-example-app
```
Comment (mosesn) [inline source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md]: https

File: source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+[argonaut]: http://argonaut.io
+[circe]: http://circe.io
+[swagger]: http://swagger.io
+[swaggerpetstore]: http://petstore.swagger.io
+[mustache]: http://mustache.github.io/
```
Comment (mosesn) [inline source/blog/2016-03-01-reusable-http-contracts-for-finagle-with-fintrospect.md]: https

**Comment by mosesn** [comment]
LGTM except for a few nits.  Thanks for the great blog post!

**Comment by vkostyukov** [comment]
+1 to squashing the commits. Thanks @daviddenton - this looks pretty great!

**Comment by daviddenton** [comment]
Hi,

I've opened another PR with a single commit for this.. just checking if is
anyone waiting for me to make any other changes before this is merged.

:)

/dave

**Pull request closed without merging**